static/app/views/externalRedirect/index.spec.tsx

```diff
@@ -0,0 +1,62 @@
+import {act, render, screen} from 'sentry-test/reactTestingLibrary';
+
+import ExternalRedirect from 'sentry/views/externalRedirect';
+
+jest.mock('sentry/utils/useLocation', () => ({
+  useLocation: jest.fn(),
+}));
+
+const mockedUseLocation = jest.requireMock('sentry/utils/useLocation').useLocation;
+
+describe('ExternalRedirect', () => {
+  let originalHref;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.useFakeTimers();
+    originalHref = window.location.href;
+    Object.defineProperty(window.location, 'href', {
+      writable: true,
+      value: 'http://initial-url.com',
+    });
+  });
+
+  afterEach(() => {
+    Object.defineProperty(window.location, 'href', {
+      writable: true,
+      value: originalHref,
+    });
+  });
+
+  it('should render correctly with a valid URL', () => {
+    const testUrl = 'http://test-url.com';
+    mockedUseLocation.mockReturnValue({query: {url: testUrl}});
+    render(<ExternalRedirect />);
+
+    // Ensure url is rendered on the redirect page
+    expect(screen.getByText(testUrl)).toBeInTheDocument();
+
+    act(() => {
+      jest.advanceTimersByTime(5000);
+    });
+
+    expect(window.location.href).toBe(testUrl);
+  });
+
+  // it('renders correctly with an invalid URL', () => {
+  //   const testUrl = 'bad://test-url.com';
+  //   mockedUseLocation.mockReturnValue({query: {url: testUrl}});
+  //   render(<ExternalRedirect />);
+
+  //   // Ensure url is rendered on the redirect page
+  //   expect(screen.getByText('Error: Invalid URL')).toBeInTheDocument();
+
+  //   act(() => {
+  //     jest.advanceTimersByTime(5000);
+  //   });
+
+  //   // expect(window).toBeNull();
+  // });
```

Comment:
Since window closes, I was having a hard time mocking this expected behavior after the 5 second timeout.

Reply:
Do we need any other tests than the main two above? @malwilley

Reply:
No, these two look good!

To test the window closing, I would use a spy to see if the function was called:

```js
const windowCloseSpy = jest.spyOn(window, 'close');

// render and stuff

expect(windowCloseSpy).toHaveBeenCalled(); 
```